src/main/java/com/gitblit/manager/FederationManager.java

```diff
@@ -363,6 +363,10 @@ public boolean accept(File file) {
 							&& file.getName().toLowerCase().endsWith(Constants.PROPOSAL_EXT);
 				}
 			});
+			if (files == null) {
```

Comment:
Your proposed change is good.  Returning an empty list is what I want, not throwing an NPE.